### PR TITLE
feat: add rhoai overlay.

### PR DESCRIPTION
**Pull request by HumairAK**
## The issue resolved by this Pull Request:
Related: https://issues.redhat.com/browse/RHOAIENG-2877

## Description of your changes:
Provide a rhoai overlay alongside an odh overlay
This will allow us to patch any rhoai specific changes in the future
Also add argo server image to params.env so odh operator can sub this in for downstream.

One change here to watch out is the change to the configmapGenerator suffix, this makes it so that the 2 configmaps created for the DSPO deployment no longer have the hash suffix.

## Testing instructions
* confirm kustomize builds in the overlay directories build as expected, and that argo image is subbed in
* `make deployODH` behavior should not have changed other than the configmap name change, the output should be otherwise the same 
* currently the rhoai overlay should function identically to the odh overlay

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-572`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/572/head
git checkout -b pullrequest ef760ea0e1c3f4bb2f912df0dec64fcbc36b527b
oc new-project opendatahub
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-572"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-572`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-572`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-572`

File: .github/resources/argo-lite/patch/argo_server_deployment.yaml [hunk 1]
```diff
@@ -14,4 +14,4 @@ spec:
               memory: 500Mi
             requests:
               cpu: 20m
-              memory: 100m
+              memory: 100Mi
```
Comment (gregsheremeta) [inline .github/resources/argo-lite/patch/argo_server_deployment.yaml:17]: seems unrelated -- extract to a separate PR?

File: config/overlays/make-argodeploy/kustomization.yaml [hunk 2]
```diff
@@ -4,6 +4,10 @@ namespace: opendatahub
 resources:
 - ../../argo
 
+# This patch is to override the `$(IMAGES_ARGO_SERVER)` param in config/argo
+# It is then overwridden by the `images` field further down, allowing us to control the
```
Comment (gregsheremeta) [inline config/overlays/make-argodeploy/kustomization.yaml:8]: control the ... what?

**Review by gregsheremeta — COMMENTED** [review]
I've forgotten what little I used to know about kustomize, so I'm not a great help with reviewing the logic.

Was there any angst about adding an "rhoai" directory in this ODH repo? Have we done that in other places?

File: config/overlays/make-argodeploy/kustomization.yaml (hunk 2 above)
Comment (HumairAK) [reply config/overlays/make-argodeploy/kustomization.yaml]: updated

File: .github/resources/argo-lite/patch/argo_server_deployment.yaml (hunk 1 above)
Comment (HumairAK) [reply .github/resources/argo-lite/patch/argo_server_deployment.yaml:17]: it's _kinda_ related, because we updated argo's deployment in the ci, and this unit was reporting errors

it should definitely be a new commit, so I added it as such

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-572`

**Comment by HumairAK** [comment]
>Was there any angst about adding an "rhoai" directory in this ODH repo? Have we done that in other places?

I don't see why there would be, the point of overlays is to configure your common yamls for different environments/platforms/use cases etc.

The existence of this folder actually helps to make sure we only put any rhoai specific manifests in this overlay, while keeping our generic templates platform agnostic

**Review by DharmitD — COMMENTED** [review]
Will let Greg review the changes he requested, but otherwise 
/lgtm
